### PR TITLE
feat: Improve cache

### LINKED ISSUE #862: Improve Cache logic on Android
### Description

We should change two things in our cache engine:

1 - Currently our cache does not limit the disk cache size as we do with memory. Create extra config parameter to limit it and create logic to delete most oldest cache if limit is reached.
2 - Join disk/memory expire time logic to be just one thing, because actually the cache expires only memory but it continues living on disk. And to make things more clear, we should change our logic to invalidate disk and memory if it goes out of time.

### PR DESCRIPTION
**Pull request by paulomeurerzup**
Signed-off-by: paulomeurerzup <paulo.meurer@zup.com.br>

### Related Issues

Closes #862 

### Description and Example

Changes beagle cache to clear and limit disk cache.

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

File: android/beagle/src/main/java/br/com/zup/beagle/android/store/DatabaseLocalStore.kt [hunk 1]
```diff
@@ -66,6 +66,38 @@ internal class DatabaseLocalStore(
...
 
+    override fun delete(key: String) {
+        database.delete(ScreenEntry.TABLE_NAME, "${ScreenEntry.KEY_COLUMN_NAME}=?", arrayOf(key))
+    }
+
+    override fun getAll(): Map<String, String> {
+        val columnsToReturn = arrayOf(ScreenEntry.KEY_COLUMN_NAME, ScreenEntry.VALUE_COLUMN_NAME)
+        val columnsForWhereClause = ""
+        val valuesForWhereClause = arrayOf<String>()
+        val cursor = database.query(
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/store/DatabaseLocalStore.kt:77]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/aab1855c10e1435ef93f08b41c3844b4f108ebd5/android/beagle/src/main/java/br/com/zup/beagle/android/store/DatabaseLocalStore.kt#L77"></span>This <code>Cursor</code> should be freed up after use with <code>#close()</code></td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | Classname | Time|
--- | --- | ---|
save_should_add_new_data_on_database | br.com.zup.beagle.android.store.DatabaseLocalStoreTest | 0.531|
save_should_log_error_when_return_value_is_minus_1 | br.com.zup.beagle.android.store.DatabaseLocalStoreTest | 0.182|


<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt [hunk 2]
```diff
@@ -27,53 +27,75 @@ import br.com.zup.beagle.android.view.ScreenRequest
...
 internal data class BeagleCache(
-    val isHot: Boolean,
     val hash: String,
-    val json: String
-)
+    val json: String,
+    val maxTime: Long,
+    val cachedTime: Long
+) {
+    fun isHot(): Boolean {
```
Comment (uziasferreirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt:41]: What is mean isHot?

Comment (uziasferreirazup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt:41]:
```suggestion
    fun isTimeExpired(): Boolean {
```

File: backend/sample/spring/src/main/resources/application.properties [hunk 3]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 #
 
-beagle.cache.endpoint.include=/sample
```
[the remark this answers is not in the document]
Comment (paulomeurerzup) [reply backend/sample/spring/src/main/resources/application.properties]: it was a mistake. Changed.

File: android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt (hunk 2 above)
Comment (paulomeurerzup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt]: changed to isExpired()

**Comment by paulomeurerzup** [comment]
This branch will be merged to release/1.2.2-android

**Pull request closed without merging**